### PR TITLE
Ex4

### PR DESCRIPTION
**Pull request by stevelee2497**

### CODE REVIEW

File: Exercise4Solution/MainActivity.cs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    {
+        private double result;
+        private double entry;
+        private Operator lastOperator;
+        private bool pointNumberFlag;
+
+        [InjectView(Resource.Id.tvResult)] private TextView tvResult;
+
+        [InjectOnClick(Resource.Id.btnNum0)]
+        private void OnClickBtnNum0(object sender, EventArgs e)
```
Comment (voquanghoa) [inline Exercise4Solution/MainActivity.cs:20]: Use only a method to handle event for all number buttons. Put the value `tag` for each button. So you can call 
```C#
AppendNumberToEntry(((Button)sender).Tag + "");
```

File: Exercise4Solution/MainActivity.cs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            var entryText = entry.ToString(CultureInfo.InvariantCulture);
+            entryText = entryText.Remove(entryText.Length - 1);
+            entry = Convert.ToDouble(entryText);
+            tvResult.Text = entryText;
+        }
+
+        [InjectOnClick(Resource.Id.btnPoint)]
+        private void OnClickBtnPoint(object sender, EventArgs e)
+        {
+            if (!pointNumberFlag)
```
Comment (voquanghoa) [inline Exercise4Solution/MainActivity.cs]: I'm not sure but you can do like this
```C#
if(!tvResult.Text.Contains("."))
{
}
```

File: Exercise4Solution/MainActivity.cs [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        private void OnClickBtnChangeStatus(object sender, EventArgs e)
+        {
+            entry = entry > 0 ? -entry : entry;
+            tvResult.Text = entry + "";
+        }
+
+        [InjectOnClick(Resource.Id.btnSquareRoot)]
+        private void OnClickBtnSquareRoot(object sender, EventArgs e)
+        {
+            entry = Math.Sqrt(entry);
```
Comment (voquanghoa) [inline Exercise4Solution/MainActivity.cs]: What's happens if entry<0 ?

File: Exercise4Solution/Resources/values/Strings.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<resources>
+    <string name="app_name">Calculator</string>
+    <string name="backspace"><![CDATA[<-]]></string>
+    <string name="back">Back</string>
+    <string name="sign_up">SIGN UP</string>
+    <string name="connect_with_facebook">CONNECT WITH FACEBOOK</string>
```
Comment (voquanghoa) [inline Exercise4Solution/Resources/values/Strings.xml]: WTF

File: Exercise4Solution/DoubleExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Exercise4Solution
+{
+    public static class DoubleExtensions
```
Comment (voquanghoa) [inline Exercise4Solution/DoubleExtensions.cs:3]: Unused ?

File: Exercise4Solution/Cheeseknife.cs [hunk 6]
```diff
@@ -0,0 +1,539 @@
+/*
```
Comment (voquanghoa) [inline Exercise4Solution/Cheeseknife.cs:1]: Why should you add this file?

File: Exercise4Solution/MainActivity.cs [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+            }
+            entry = 0;
+            pointNumberFlag = false;
+            lastOperator = o;
+            tvResult.Text = Convert.ToString(result, CultureInfo.InvariantCulture);
+        }
+
+        private void AppendNumberToEntry(string number)
+        {
+            entry = Convert.ToDouble(pointNumberFlag ? tvResult.Text + number : entry + number);
```
Comment (voquanghoa) [inline Exercise4Solution/MainActivity.cs]: I think in view result mode if the user presses a number, we should clear the result first.

File: Exercise4Solution/DoubleExtensions.cs (hunk 5 above)
Comment (stevelee2497) [reply Exercise4Solution/DoubleExtensions.cs:3]: My bad, forgot to delete

File: Exercise4Solution/Cheeseknife.cs (hunk 6 above)
Comment (stevelee2497) [reply Exercise4Solution/Cheeseknife.cs:1]: Arcording to https://github.com/MarcelBraghetto/Cheeseknife : 

There are a couple of ways to include Cheeseknife in your own Xamarin.Android project:

Option 1: Copy the Cheeseknife library project from the demo solution and add it as a reference to your Android project.
Option 2: Copy the Cheeseknife.cs source file into your own project and add it into your solution.

anyway, can you suggest any other ways to include Cheeseknife to my project ?

File: Exercise4Solution/Resources/values/Strings.xml (hunk 4 above)
Comment (stevelee2497) [reply Exercise4Solution/Resources/values/Strings.xml]: my bad, I forgot to delete

File: Exercise4Solution/Cheeseknife.cs (hunk 6 above)
Comment (voquanghoa) [reply Exercise4Solution/Cheeseknife.cs:1]: Thế đéo nào 😞

**Pull request merged**